backend/src/ds/grpc.rs

```diff
@@ -233,6 +236,41 @@ impl<Qep: QsConnector> DeliveryService for GrpcDs<Qep> {
             Status::internal("failed to create group")
         })?;
 
+        // Extract user id
+        let members = group.members().collect::<Vec<_>>();
+
+        if members.len() > 1 {
+            error!(members = %members.len(), "group must have exactly one member");
+            return Err(Status::invalid_argument(
+                "group must have exactly one member",
+            ));
+        }
+
+        let own_leaf = members.first().ok_or_else(|| {
+            error!("group must have exactly one member");
+            Status::invalid_argument("group must have exactly one member")
+        })?;
+        let credential =
+            ClientCredential::tls_deserialize_exact_bytes(own_leaf.credential.serialized_content())
+                .map_err(|_| Status::invalid_argument("invalid credential"))?;
+        let user_id = credential.identity().uuid();
+
+        // Configure the rate-limiting
+        let rl_key = RLKey::new("ds", "reserve_group_id", &["user_id", &user_id.to_string()]);
```

Comment:
Nit: could you please rename `user_id` to `user_uuid`? We use this naming since the last rename consistently everywhere now. Thank you.

Reply:
It is a little bit unfortunate that we have to allocate the string here, but I don't see a different way with the slice interface.

Reply:
This bothered me as well. I just turned everything into `&[u8]` now.